### ISSUE TITLE
Support for Gateway API metrics

### ISSUE THREAD

**Issue by mrueg** [kubernetes/kube-state-metrics#2449]
**What would you like to be added**:
Expose metrics for objects that are part of the Gateway API https://gateway-api.sigs.k8s.io/implementations/v1.1/
**Why is this needed**:

To see what objects are in the data path for the new way of doing ingress.

**Describe the solution you'd like**

* Use Custom Resource State Definition similar to HPA 
or
* Implement them as native resources in Kubernetes.

**Additional context**
-

**Comment by gaffneyd4** [comment]
Given that the project is [not planning on supporting non-native resources](https://github.com/kubernetes/kube-state-metrics/pull/2452#issuecomment-2532745890), I think this issue should be updated to request documentation showing how an end user can craft a custom resource state definition to collect metrics from gateway resources.

**Comment by gaffneyd4** [comment]
It looks like the work to describe how to collect gateway api resource metrics has been done by this project...
https://github.com/Kuadrant/gateway-api-state-metrics/blob/main/config/kuadrant/custom-resource-state.yaml

**Comment by miqm** [comment]
@gaffneyd4 @rexagod I would like to request to re-open this issue. Gateway API is a native Kubernetes resource. What's more - the default Kubernetes' Ingress NGINX is being marked as deprecated and it's suggested to migrate to Gateway API (https://kubernetes.io/blog/2025/11/11/ingress-nginx-retirement/), Therefore native support of `gateway.networking.k8s.io/v1` is most welcome.

**Comment by spiarh** [comment]
I agree with @miqm, especially since the deprecation of ingress-nginx, many people are currently using this opportunity to migrate to Gateway API (mainly HTTProute).
